infra/modules/azure_storage_account/variables.tf

```diff
@@ -0,0 +1,175 @@
+    user_assigned_identity_id = optional(string, null)
+    key_vault_key_id          = optional(string, null)
+    managed_hsm_key_id        = optional(string, null)
+  })
+  description = "(Optional) Customer managed key to use for encryption"
+  default     = { enabled = false }
+}
+
+variable "force_public_network_access_enabled" {
+  type        = bool
+  description = "(Optional) Whether the Storage Account permits public network access or not. Defaults to false."
+  default     = false
+}
+
+variable "access_tier" {
+  type        = string
+  description = "(Optional) Access tier of the Storage Account. Defaults to Hot."
+  default     = "Hot"
+}
+
+variable "subservices_enabled" {
+  type = object({
+    blob  = optional(bool, true)
+    file  = optional(bool, false)
+    queue = optional(bool, false)
+    table = optional(bool, false)
+  })
+  description = "(Optional) Subservices enabled for the Storage Account. Creates peps for enabled services. By default, only blob is enabled."
+}
+
+variable "blob_features" {
+  type = object({
+    restore_policy_days   = optional(number, 0)
+    delete_retention_days = optional(number, 0)
+    last_access_time      = optional(bool, false)
+    versioning            = optional(bool, false)
+    change_feed = object({
+      enabled           = optional(bool, false)
+      retention_in_days = optional(number, 0)
+    })
+    immutability_policy = object({
+      enabled                       = optional(bool, false)
+      allow_protected_append_writes = optional(bool, false)
+      period_since_creation_in_days = optional(number, 730)
+    })
+  })
+  description = "(Optional) Blob features configuration"
+  default = {
+    restore_policy_days   = 0
+    delete_retention_days = 0
+    last_access_time      = false
+    versioning            = false
+    change_feed           = { enabled = false, retention_in_days = 0 }
+    immutability_policy   = { enabled = false }
+  }
+
+  # https://learn.microsoft.com/en-us/azure/storage/blobs/point-in-time-restore-overview#limitations-and-known-issues
+  validation {
+    condition     = (var.blob_features.immutability_policy.enabled == true && var.blob_features.restore_policy_days == 0) || var.blob_features.immutability_policy.enabled == false
+    error_message = "Immutability policy doesn't support Point-in-Time restore"
+  }
+
+  validation {
+    condition     = var.blob_features.delete_retention_days == 0 || (var.blob_features.delete_retention_days >= 1 && var.blob_features.delete_retention_days <= 365)
+    error_message = "Delete retention days must be 0 to disable the policy or between 1 and 365."
+  }
+
+  validation {
+    condition     = var.blob_features.restore_policy_days == 0 || (var.blob_features.restore_policy_days >= 1 && var.blob_features.restore_policy_days <= 365)
+    error_message = "Restore policy days must be 0 to disable the policy or between 1 and 365."
+  }
+}
+
+variable "network_rules" {
```

Comment:
Not sure if introducing this variable is valuable or not.
I mean, the storage account provides peps already, why does anyone need to stick to firewall rules?

Reply:
I agree, in normal cases it shouldn't be needed. I added it for retro compatibility for storage accounts that need to enable public access.
What do you think?

Reply:
ok, for me it can stay